src/modules/function/invocation.rs

```diff
@@ -74,6 +74,22 @@ impl SyntaxModule<ParserMetadata> for FunctionInvocation {
                 };
             }
             let function_unit = meta.get_fun_declaration(&self.name).unwrap().clone();
+            let expected_arg_count = function_unit.arg_refs.len();
+            let actual_arg_count = self.args.len();
+            let optional_count = function_unit.arg_optionals.len();
+
+
+            if actual_arg_count < expected_arg_count { //there are missing arguments
+                //check if we can compensate with optional arguments stored in fun_unit
+                if actual_arg_count >= expected_arg_count - optional_count {
+                    let missing = expected_arg_count - actual_arg_count;
+                    let provided_optional = optional_count - missing;
+                    for exp in function_unit.arg_optionals.iter().skip(provided_optional){
+                        self.args.push(exp.clone());
+                    }
+                }
```

Comment:
No error when there are not enough arguments? Or is it handled somewhere else?

Reply:
Thanks for the corrections.
Yes it'll be handled somewhere else, in ```run_function_with_args```. 
On that note, when there aren't enough arguments, should the error message display the expected number of arguments including the optional arguments or without the optional arguments?  

I didn't think abot refs. Should an optional argument never be allowed to be a ref?

Thanks.

Reply:
That's a good question!

1. I think that it should rather display the minimum required arguments and perhaps the total amount of arguments in the parentheses
2. Refs shouldn't be allowed as optional arguments at least for now since there is nothing like a `nullptr` in Amber and I don't think it would be necessary

Thank you so much for this PR! I appreciate your work 🙏

Reply:
I've gone ahead and made those two changes, let me know what you think!
Thank you.